libraries/shared/watcher/storage_watcher.go

```diff
@@ -100,30 +103,41 @@ func (storageWatcher StorageWatcher) Execute(queueRecheckInterval time.Duration)
 	}
 }
 
-func (storageWatcher StorageWatcher) getTransformer(diff utils.StorageDiff) (transformer.StorageTransformer, bool) {
+func (storageWatcher StorageWatcher) getTransformer(diff utils.PersistedStorageDiff) (transformer.StorageTransformer, bool) {
 	storageTransformer, ok := storageWatcher.KeccakAddressTransformers[diff.HashedAddress]
 	return storageTransformer, ok
 }
 
-func (storageWatcher StorageWatcher) processRow(diff utils.StorageDiff) {
-	storageTransformer, isTransformerWatchingAddress := storageWatcher.getTransformer(diff)
+func (storageWatcher StorageWatcher) processRow(diffInput utils.RawStorageDiff) {
+	diffID, err := storageWatcher.StorageDiffRepository.CreateStorageDiff(diffInput)
+	if err != nil {
+		if err == repositories.ErrDuplicateDiff {
+			logrus.Info("ignoring duplicate diff")
+			return
+		}
+		logrus.Warnf("failed to persist storage diff: %s", err.Error())
+		// TODO: bail? Should we continue attempting to transform a diff we didn't persist
```

Comment:
this is a good question 🤔 on one hand I'd want to bail so we know there's a problem, but on the other hand I'd want to keep processing so we don't lose the diff.

Reply:
Yeah, I can't really think of what would cause this error other than some sort of ephemeral problem connecting to the DB. Thinking that we should probably at least try and continue processing, but also watch the logs to see if this happens and potentially impose some sort of retries.